### DISCUSSION TITLE
[Dashboards - Feature Request] Link to Row within long dashboard

### DISCUSSION THREAD

**Discussion by samjewell** [grafana/grafana#58605] [Dashboards, Panels & Visualizations]
**What happened**:
I want to create a link and share it, where the link destination is a **single row** within a very long dashboard. For example [this Billing/Usage dashboard](https://play.grafana.org/d/vXjDa4IVk/billing-usage?orgId=1) is very long, and sometimes I want to link directly to a single row which is way down the dashboard and certainly below the fold, such as "Traces Instance Stats".

I fact I often I want to link to that row from summary content which is placed at the top of the dashboard above the fold, in order to more quickly and easily "drill" into what's behind that summary data.

But this doesn't seem to be possible. I inspected the source code for the dashboard, but can't see any `id` HTML Attr on the DOM elements that make up the row-heading:
![image](https://user-images.githubusercontent.com/2903904/200817787-44ce516c-31e3-406d-94f3-784392450579.png)

**What you expected to happen**:
As a minimum, I want:
- an `id` HTML elem on the Row, so I can link to it
- For a page load of this URL: https://play.grafana.org/d/vXjDa4IVk/billing-usage#traces-instance-stats to retain the anchor in the URL after page-load (and not remove it, as seems to be happening currently), and also to scroll the page down to that row when it renders.

**How to reproduce it (as minimally and precisely as possible)**:
See above

**Anything else we need to know?**:
I realise this is a feature request, but the Github Discussions I've created on this repo 1 year ago still haven't received a response.

**Environment**:
- Grafana version: 
![image](https://user-images.githubusercontent.com/2903904/200818419-e1cbbd63-da63-4b5b-b619-6b7826579d5c.png)

- OS Grafana is installed on: GrafanaCloud
- User OS & Browser: MacOS 12.6.1, Chrome 107.0.5304.87 (Official Build) (arm64)

**Comment by axelavargas** [comment]
Hi @samjewell 👋🏾, yes, this falls into the feature request category 😄. I personally agree that this will be a cool feature to have but unfortunately, our roadmap is a bit stretched at the moment.
(🤔 _I am thinking we can consider this when implementing [the new framework and runtime for dashboards. ](https://github.com/grafana/grafana/issues/52484)._)

I'm going to turn this issue into a GH discussion to follow our process, but this issue will be referenced in the epic for future consideration 😄

Thank you 🙏🏾

**Comment by Itzikg1** [comment]
Hi @axelavargas 
Could you please update on this? this feature will be very useful for us.

Thank you ! 🙏🏾

**Comment by clompsy** [comment]
I agree this would be very useful, but would add one important detail:
If the linked row isn't open (collapsed), it shall be opened (expanded).
We have dashboard with an overview with lots of closed rows (to avoid loading data eagerly). Depending on the interest the users can expand different rows to add more details for their view. I'd like to be able to use Data Links in visualizations to automatically scroll to and open a row that contains details on the clicked element.
It's nothing of urgency of course, just a convenience thing to keep bigger dashboards more intuitive without the need to link to other dashboards (which would be the current approach to solve this).

Thank you :)

**Reply by timo1707** [reply]
+1

**Reply by sandrovw64** [reply]
+1

**Comment by sandrovw64** [comment]
This would be really useful for long dashboards, and also could be done on TV mode, playlists

**Comment by skeagy** [comment]
+1

My use case is a table in first row with summary info and a few sparklines, and I'd like to click in that summary table to a specific row later in the dashboard with more detailed panels. I landed in this discussion trying to figure out how to do exactly this.

+1 Automatically opening the target row if it is closed would be an excellent improvement to save dashboard load time and enable larger scale use.

![use_case_for_links_to_specific_dashboard_rows](https://github.com/grafana/grafana/assets/23060915/1c032985-d458-4e05-a4fd-b2f9134b13ad)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by samjewell** [comment]
I have created an issue for this, seeing as there's been plenty of interest, and I still want/need it. It's moved to here:
#82827 

The Github UI doesn't make this especially easy to see, but you can find it in the right-nav here:

![image](https://github.com/grafana/grafana/assets/2903904/af6aaa3c-c1a9-48b0-8f7c-45f03605043b)
